src/core_functions/scalar/list/array_slice.cpp

```diff
@@ -177,16 +389,29 @@ static unique_ptr<FunctionData> ArraySliceBind(ClientContext &context, ScalarFun
 		throw BinderException("ARRAY_SLICE can only operate on LISTs and VARCHARs");
 	}
 
-	return make_uniq<VariableReturnBindData>(bound_function.return_type);
+	bool start_is_throw = false;
+	if (!arguments[1]->IsFoldable()) {
+		start_is_throw = true;
```

Comment:
What happens if the start is foldable but not a constant, e.g. `(-9223372036854775807)-1`?